### PR TITLE
Changed the hidden_size to d_model for XLNET docs

### LINKED ISSUE #11938: [docs] XLNETModel forward returns last_hidden_state 3rd dim should be d_model instead of hidden_size
**Link to doc** - https://huggingface.co/transformers/model_doc/xlnet.html#xlnetmodel

**Problem description** - forward method of XLNETModel returns `last_hidden_state` with dimension `(batch_size, num_predict, hidden_size)`. However, `hidden_size` is not an XLNET config (unlike for BERT) instead that dimension should be `d_model` which is a config of XLNETModel.

**current shape** - (batch_size, num_predict, hidden_size)
**expected shape** - (batch_size, num_predict, d_model)

## Environment info
Not required

### Who can help
@sgugger
<!-- Your issue will be replied to more quickly if you can figure out the right person to tag with @
 If you know how to use git blame, that is the easiest way, otherwise, here is a rough guide of **who to tag**.
 Please tag fewer than 3 people.

### PR DESCRIPTION
**Pull request by Muktan**
# What does this PR do?
Changes the use of hidden_size to d_model as XLNET model (unlike BERT and other) uses `d_model` instead of `hidden_size`.

Fixes #11938 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).

## Who can review?
@sgugger @NielsRogge

### CODE REVIEW

**Comment by sgugger** [comment]
Many models use different names for this and `hidden_size` is a constant property that works for all of them, so we will keep the docstrings updated with this. It also simplifies copy-pasting the docstrings between models.

**Comment by Muktan** [comment]
> Many models use different names for this and `hidden_size` is a constant property that works for all of them, so we will keep the docstrings updated with this. It also simplifies copy-pasting the docstrings between models.

Okay, in that case at least we can mention somewhere in the documentation of the XLNET that hidden_size is same as d_model. Because for someone who starts working with XLNET and has not worked with other models won't understand what is hidden_size, as hidden_size is not described in XLNET docs and is mentioned as one of the dimensions in the output.

**Comment by sgugger** [comment]
How about adding an entry in the glossary page, along with input IDs, attention mask etc?

**Comment by Muktan** [comment]
> How about adding an entry in the glossary page, along with input IDs, attention mask etc?

along with.. means inside those sections [input IDs section, attention mask section] or creating another section for it?

If thinking of adding it to the glossary we can write a line in docs of XLNET, BERT ..(other models) that explanation of some terms can be explored in the glossary. As all documentations (pandas, pytorch... etc) don't have glossary section and it can't be expected from the reader that he/she will search for glossary section if they encounter any term not explain in a documentation page.
